Content.Client/Explosion/ExplosionOverlay.cs

```diff
@@ -0,0 +1,124 @@
+using JetBrains.Annotations;
+using Robust.Client.Graphics;
+using Robust.Shared.Enums;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Client.Explosion;
+
+[UsedImplicitly]
+public sealed class ExplosionOverlay : Overlay
+{
+    /// <summary>
+    ///     The explosion that needs to be drawn. This explosion is currently being processed by the server and
+    ///     expanding outwards.
+    /// </summary>
+    internal Explosion? ActiveExplosion;
```

Comment:
if it can only draw one explosion at once what does having multiple explosions at once look like client-wise?

Reply:
It can draw more than one at a time fine, but if an explosion is big enough that the server processes it over several ticks, it currently only processes that one explosion at a time. This is the "active explosion", and the visual overlay for it should be synced up with the area that the server has damaged.

Reply:
aight makes sense